.github/scripts/curate_autogenerated_schema.py

```diff
@@ -0,0 +1,1999 @@
+                    },
+                    schema_dict["definitions"][type_definition_key]["properties"]["type"]
+                ]
+            }
+        else:
+            pattern_properties_regex_match_ref = {
+                "oneOf": [
+                    {
+                        "$ref": property_old.get("items", {}).get("$ref")
+                    },
+                    {
+                        "type": "string"
+                    }
+                ]
+            }
+
+        # For situations where the type is defined by a either a single string or an array of items where
+        # Each element of the array is equivalent to the 'type' property
+        if allow_type_array_as_only_input:
+            pattern_properties_regex_match_ref["oneOf"].append(
+                {
+                    "type": "array",
+                    "items": {
+                        "oneOf": [
+                            {
+                                "type": "string"
+                            },
+                            {
+                                "$ref": f"#/definitions/{type_definition_key}"
+                            }
+                        ]
+                    }
+                }
+            )
+
+        # Re-initialise the definitions
+        schema_dict["definitions"][definition_key]["properties"][property_name] = {
+            "description": property_old.get("description", ""),
+            "oneOf": [
+                {
+                    "items": property_old.get("items", {}),
+                    "type": property_old.get("type", {})
+                },
+                {
+                    "patternProperties": {
+                        "^[_a-zA-Z][a-zA-Z0-9_-]*$": pattern_properties_regex_match_ref
+                    },
+                    "type": "object"
+                }
+            ]
+        }
+    else:
+        # For WorkflowStep, we just allow the patternProperties to match the property_old items type
+        # While the property itself is an object
+        # Re-initialise the definitions
+        schema_dict["definitions"][definition_key]["properties"][property_name] = {
+            "description": property_old.get("description", ""),
+            "oneOf": [
+                {
+                    "items": property_old.get("items", {}),
+                    "type": property_old.get("type", {})
+                },
+                {
+                    "patternProperties": {
+                        "^[_a-zA-Z][a-zA-Z0-9_-]*$": property_old.get("items", {})
+                    },
+                    "type": "object"
+                }
+            ]
+        }
+
+    return schema_dict
+
+
+def fix_secondary_file_schema(schema_dict: Dict) -> Dict:
+    """
+    SecondaryFileSchema should allow for a single string entry to represent the secondary file pattern
+    Although not recommended, it is allowed in the CWL spec
```

Comment:
Why is it not recommended?

Reply:
Is this okay to keep in or should I remove the commentary?